nxc/protocols/smb/kerberos.py

```diff
@@ -0,0 +1,381 @@
+        #             print(hexlify(nthash).decode())
+        #     if not aesKey:
+        #         salt = domain.upper() + username
+        #         aesKey = _AES256CTS.string_to_key(password, salt, params=None).contents
+        #         if logging.getLogger().level == logging.DEBUG:
+        #             logging.debug('AESKey')
+        #             print(hexlify(aesKey).decode())
+
+        # # Get the encrypted ticket returned in the TGS. It's encrypted with one of our keys
+        # cipherText = tgs['ticket']['enc-part']['cipher']
+
+        # # Check which cipher was used to encrypt the ticket. It's not always the same
+        # # This determines which of our keys we should use for decryption/re-encryption
+        # newCipher = _enctype_table[int(tgs['ticket']['enc-part']['etype'])]
+        # if newCipher.enctype == Enctype.RC4:
+        #     key = Key(newCipher.enctype, nthash)
+        # else:
+        #     key = Key(newCipher.enctype, aesKey)
+
+        # # Decrypt and decode the ticket
+        # # Key Usage 2
+        # # AS-REP Ticket and TGS-REP Ticket (includes tgs session key or
+        # #  application session key), encrypted with the service key
+        # #  (section 5.4.2)
+        # plainText = newCipher.decrypt(key, 2, cipherText)
+        # encTicketPart = decoder.decode(plainText, asn1Spec=EncTicketPart())[0]
+
+        # # Print the flags in the ticket before modification
+        # logging.debug('\tService ticket from S4U2self flags: ' + str(encTicketPart['flags']))
+        # logging.debug('\tService ticket from S4U2self is'
+        #                 + ('' if (encTicketPart['flags'][TicketFlags.forwardable.value] == 1) else ' not')
+        #                 + ' forwardable')
+
+        # # Customize flags the forwardable flag is the only one that really matters
+        # logging.info('\tForcing the service ticket to be forwardable')
+        # # convert to string of bits
+        # flagBits = encTicketPart['flags'].asBinary()
+        # # Set the forwardable flag. Awkward binary string insertion
+        # flagBits = flagBits[:TicketFlags.forwardable.value] + '1' + flagBits[TicketFlags.forwardable.value + 1:]
+        # # Overwrite the value with the new bits
+        # encTicketPart['flags'] = encTicketPart['flags'].clone(value=flagBits)  # Update flags
+
+        # logging.debug('\tService ticket flags after modification: ' + str(encTicketPart['flags']))
+        # logging.debug('\tService ticket now is'
+        #                 + ('' if (encTicketPart['flags'][TicketFlags.forwardable.value] == 1) else ' not')
+        #                 + ' forwardable')
+
+        # # Re-encode and re-encrypt the ticket
+        # # Again, Key Usage 2
+        # encodedEncTicketPart = encoder.encode(encTicketPart)
+        # cipherText = newCipher.encrypt(key, 2, encodedEncTicketPart, None)
+
+        # # put it back in the TGS
+        # tgs['ticket']['enc-part']['cipher'] = cipherText
```

Comment:
Is this for future use? Or old code that could be removed?

If this is for future implementation I think we could add a comment at the top clarifying it.

Reply:
removed